### DISCUSSION TITLE
Add new tab in allure report

### DISCUSSION THREAD

**Discussion by mwalaszkowski** [allure-framework/allure2#2293] [Documentation] [answered]
Hi all

I want to add new custom tab in allure report - tab with some custom graphs, tables and information witch my tests generate.
Is this well documented somewhere or maybe some of you have some experience how to create custom tab in report? For now i just know that i should create custom plugin, but still i don't really know how to start, build etc.

Thanks for all replies.

**Comment by baev** [comment, marked as answer]
There isn't much information about custom plugins available. The main reason is that historically, Allure was written in both Java & JavaScript with some technologies that can be considered outdated today. So, it takes a lot of work to get it started. Besides, there is no easy way to distribute & manage plugins, so the only option is to manage custom allure-commandline distribution and bundle custom plugins. 

We do have plugins to rewrite all the code and change the technologies used, but this is a completely different story. 

Having all this in mind, the most convenient way to create a custom tab is to fork the repository and follow the code examples in the plugins folder. 

However, some of the information on Plugin API can be found in old documentation: https://github.com/allure-framework/allure-docs/tree/main/content/plugins

**Comment by baev** [comment]
The second reply with a bit more optimistic voice ;)

Our team actively developing the new test reporting platform (known as Allure 3) with extensibility in mind. Feel free to share ideas on why you need a custom report; we would love to hear that.

**Comment by mwalaszkowski** [comment]
Thanks for reply, this link can be very useful! 
Is there any Beta version of allure 3 reporter framework? Can you provide me a link where I can post about my ideas?

**Reply by baev** [reply]
> Can you provide me a link where I can post about my ideas?

https://github.com/orgs/allure-framework/discussions/categories/ideas-feedback

> Is there any Beta version of allure 3 reporter framework?

not yet

**Comment by mwalaszkowski** [comment]
Thanks for reply. We can close discussion.
